pkg/controller/controlplane/actuator.go

```diff
@@ -17,37 +17,54 @@ package controlplane
 import (
 	"context"
 	"fmt"
-	"strconv"
 	"time"
 
 	"github.com/go-logr/logr"
-	"k8s.io/client-go/util/retry"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
 
 	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
 	"github.com/gardener/gardener/extensions/pkg/controller/controlplane"
+	controllererror "github.com/gardener/gardener/extensions/pkg/controller/error"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
-	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/controllerutils"
 	azurev1alpha1 "github.com/gardener/remedy-controller/pkg/apis/azure/v1alpha1"
 )
 
+const (
+	// GracefulDeletionWaitInterval is the default interval for retry operations.
+	GracefulDeletionWaitInterval = 1 * time.Minute
+	// GracefulDeletionTimeout is the timeout that defines how long the actuator should wait for remedy controller resources to be deleted
+	// gracefully by the remedy controller itself
+	GracefulDeletionTimeout = 10 * time.Minute
+)
+
+// TimeNow returns the current time. Exposed for testing.
+var TimeNow = time.Now
```

Comment:
This appears unused, is it really needed? If it is, the proper way to switch between "real time" and "testing time" would be via `clock.Clock` (could be added as a field to the actuator), see g/g code.

Reply:
It is not needed.